virtualsmartcard/win32/BixVReader/PipeReader.cpp

```diff
@@ -217,22 +230,28 @@ DWORD PipeReader::startServer() {
 				OutputDebugString(L"[BixVReader]Pipe data");
 				if (command==0)
 					powered=0;
-				if (command==0 && waitRemoveIpr!=NULL) {// card removed
+				if (command==0 && !waitRemoveIpr.empty()) {
+					// card removed
 					SectionLocker lock(device->m_RequestLock);
 					state=SCARD_ABSENT;
-					if (waitRemoveIpr->UnmarkCancelable()==S_OK) {						
-						waitRemoveIpr->CompleteWithInformation(STATUS_SUCCESS, 0);
+					while (!waitRemoveIpr.empty()) {
+						CComPtr<IWDFIoRequest> ipr = waitRemoveIpr.back();
+						if (ipr->UnmarkCancelable()==S_OK)
+							ipr->CompleteWithInformation(STATUS_SUCCESS, 0);
+						waitRemoveIpr.pop_back();
 					}
-					waitRemoveIpr=NULL;
 				}
-				else if (command==1 && waitInsertIpr!=NULL) {// card inserted
+				else if (command==1 && !waitRemoveIpr.empty()) {
+					// card inserted
 					SectionLocker lock(device->m_RequestLock);
 					state=SCARD_SWALLOWED;
 					initProtocols();
-					if (waitInsertIpr->UnmarkCancelable()==S_OK) {
-						waitInsertIpr->CompleteWithInformation(STATUS_SUCCESS, 0);
+					while (!waitInsertIpr.empty()) {
+						CComPtr<IWDFIoRequest> ipr = waitRemoveIpr.back();

```

Comment:
Copy paste error. Should be `CComPtr<IWDFIoRequest> ipr = waitInsertIpr.back();`